Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -62,6 +65,35 @@ uint8_t SPI::inscribe(SPI::Peripheral& spi) {
     SPI::registered_spi[id] = spi_instance;
     SPI::registered_spi_by_handler[spi_instance->hspi] = spi_instance;
 
+    // Create socket for this SPI peripheral
+    int spi_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+    // Bind this socket to the local port 200x
+    sockaddr_in local_address;
+    local_address.sin_family = AF_INET;
+    local_address.sin_port = htons(2000 + id_counter);
```

Comment:
dont use magic numbers, do something like `#define SPI_PORT_BASE 2000 `and then use like `htons(SPI_PORT_BASE + id_counter)`